internal/api/policy_collection_mapping.go

```diff
@@ -53,30 +53,45 @@ type policyCollectionMappingAPI struct {
 
 // Read returns data for a policy collection mapping.
 func (a policyCollectionMappingAPI) Read(ctx context.Context, collectionUUID string, policyUUID string) (*PolicyCollectionMapping, error) {
-	var query struct {
-		PolicyCollection struct {
-			PolicyMappings struct {
-				Edges []struct {
-					Node PolicyCollectionMapping
-				}
-			}
-		} `graphql:"policyCollection(uuid: $uuid)"`
-	}
+	cursor := ""
+	for {
+		var query struct {
+			PolicyCollection struct {
+				PolicyMappings struct {
+					Edges []struct {
+						Node PolicyCollectionMapping
+					}
+					PageInfo struct {
+						HasNextPage bool
+						EndCursor   string
+					}
+					Problems []Problem
+				} `graphql:"policyMappings(first: 100, after: $cursor)"`
+			} `graphql:"policyCollection(uuid: $uuid)"`
+		}
 
-	variables := map[string]any{
-		"uuid": graphql.String(collectionUUID),
-	}
-	if err := a.c.Query(ctx, &query, variables); err != nil {
-		return nil, NewAPIError(err)
-	}
+		variables := map[string]any{
+			"uuid":   graphql.String(collectionUUID),
+			"cursor": graphql.String(cursor),
+		}
+		if err := a.c.Query(ctx, &query, variables); err != nil {
+			return nil, NewAPIError(err)
+		}
+		if err := FromProblems(ctx, query.PolicyCollection.PolicyMappings.Problems); err != nil {
+			return nil, err
+		}
 
-	for _, edge := range query.PolicyCollection.PolicyMappings.Edges {
-		if edge.Node.Policy.UUID == policyUUID {
-			return &edge.Node, nil
+		for _, edge := range query.PolicyCollection.PolicyMappings.Edges {
+			if edge.Node.Policy.UUID == policyUUID {
+				return &edge.Node, nil
+			}
 		}
-	}
 
-	return nil, NotFound{"Policy collection not found"}
+		if !query.PolicyCollection.PolicyMappings.PageInfo.HasNextPage {
+			return nil, NotFound{"Policy collection mapping not found"}
+		}
+		cursor = query.PolicyCollection.PolicyMappings.PageInfo.EndCursor
```

Comment:
@albertodonato I haven't checked but am trusting you that this DTRT when there's < 100 results in total :sweat_smile:

Reply:
yeah when there's < 100 results, `HasNextPage` is false, so it would exit earlier.